resources/qml/MainWindow/TopHeader.qml

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2018 Ultimaker B.V.
+// Cura is released under the terms of the LGPLv3 or higher.
+
+import QtQuick 2.7
+import QtQuick.Controls 1.1
+import QtQuick.Controls.Styles 1.1
+
+import UM 1.4 as UM
+import Cura 1.0 as Cura
+
+import "../Account"
+
+Rectangle
+{
+    id: base
+
+    implicitHeight: UM.Theme.getSize("topheader").height
+    implicitWidth: UM.Theme.getSize("topheader").width
+    color: UM.Theme.getColor("topheader_background")
+
+    Image
+    {
+        id: logo
+        anchors.left: parent.left
+        anchors.leftMargin: UM.Theme.getSize("default_margin").width
+        anchors.verticalCenter: parent.verticalCenter
+
+        source: UM.Theme.getImage("logo")
+        width: UM.Theme.getSize("logo").width
+        height: UM.Theme.getSize("logo").height
+
+        sourceSize.width: width
+        sourceSize.height: height
+    }
+
+    Row
+    {
+        id: stagesListContainer
+
+        anchors
+        {
+            horizontalCenter: parent.horizontalCenter
+            leftMargin: UM.Theme.getSize("default_margin").width
+        }
+
+        // The topheader is dynamically filled with all available stages
+        Repeater
+        {
+            id: stagesHeader
+
+            model: UM.StageModel { }
+
+            delegate: Button
+            {
+                text: model.name.toUpperCase()
+                checkable: true
+                checked: model.active
+                exclusiveGroup: topheaderMenuGroup
+                style: UM.Theme.styles.topheader_tab
+                height: UM.Theme.getSize("topheader").height
+                onClicked: UM.Controller.setActiveStage(model.id)
+                iconSource: model.stage.iconSource
+
+                property color overlayColor: "transparent"
+                property string overlayIconSource: ""
+            }
+        }
+
+        ExclusiveGroup { id: topheaderMenuGroup }
+    }
+
+    // Shortcut button to quick access the Toolbox
+    Button
+    {
+        id: toolboxShortcutButton
+        anchors
+        {
+            right: accountWidget.left
+            rightMargin: UM.Theme.getSize("default_margin").width
+            verticalCenter: parent.verticalCenter
+        }
+        style: ButtonStyle
+        {
+            background: Rectangle
+            {
+                color: control.hovered ? UM.Theme.getColor("secondary") : UM.Theme.getColor("topheader_button_background_active")
+                radius: 2 * screenScaleFactor
```

Comment:
If you look at the new designs, the buttons will be rounded. So we will need to do it for many components.

Reply:
The button should be extracted to a single component and be re-used everywhere. Otherwise it's impossible to keep a consistent style.

Reply:
I planned to reuse the ActionButton, but I don't remember now why I discarded the idea. I will try again tomorrow.

Reply:
Now I remember, I had some troubles to reuse the ActionButton because then I can't use `action` (or at least it didn't work), so I did it like this.

This is what I tried, but couldn't make it work:
```
Cura.ActionButton
{
    anchors
    {
        right: accountWidget.left
        rightMargin: UM.Theme.getSize("default_margin").width
        verticalCenter: parent.verticalCenter
    }
    text: catalog.i18nc("@action:button", "Toolbox")
    color: UM.Theme.getColor("main_window_header_button_background_active")
    hoverColor: UM.Theme.getColor("secondary")
    textColor: UM.Theme.getColor("main_window_header_button_text_active")
    textHoverColor: UM.Theme.getColor("main_window_header_button_text_active")
    action: Cura.Actions.browsePackages
}
```

If you have some ideas how to do it, I really prefer this way, of course.

Reply:
That looks good. What problems did you run into when trying it like this? Did you get an error? Do you know why the action didn't work?

Reply:
This is fixed. It didn't work because the ActionButton uses QtQuick Controls 2, and the `action` property is not used anymore. I just did like this: https://github.com/Ultimaker/Cura/blob/a277bd9f3b8ee3383be53e805e47058b49a7d3f9/resources/qml/MainWindow/MainWindowHeader.qml#L94